neurom/fst/_neuritefunc.py

```diff
@@ -260,19 +265,34 @@ def _seg_midpoint(sec):
     return map_segments(_seg_midpoint, neurites, neurite_type)
 
 
+def segment_path_lengths(neurites, neurite_type=NeuriteType.all):
+    '''Returns pathlengths between all non-root points and their root point'''
+    pathlength = {}
+    neurite_filter = is_type(neurite_type)
+
+    def _get_pathlength(section):
```

Comment:
Because this one is not efficient when all pathlengths need to be queried.

Reply:
Given that recursion is more expensive than iteration in python, it's hard to believe that there is a significant advantage here. On the contrary, with a deep enough morphology this could hit the recursion limit, which is not tested because test morphologies are small.